### ISSUE TITLE
`Optics` docs

### ISSUE THREAD

**Issue by nomisRev** [kategory/kategory#392]

**Comment by nomisRev** [comment]
Any1 can guide me in a good direction? Perhaps some good reads for inspiration? This is one of the best examples of Optics I know https://github.com/circe/circe/blob/master/modules/optics/src/main/scala/io/circe/optics/JsonPath.scala

**Comment by raulraja** [comment]
@nomisRev I'd encourage you to talk directly to @travisbrown and @julien-truffaut. They are some of the nicest guys I've had the chance to meet in the Scala community and may be able to point you to some examples or direction on where to get those.

**Comment by diesalbla** [comment]
@nomisRev I am not sure if this issue is still open. If it is, and you are still working on it, then the best talk about lenses I have found so far is this talk by SPJ. 

https://skillsmatter.com/skillscasts/4556-simon-peyton-jones

It may be worth having a working example of what a coding task looks like on Kotlin, with and without lenses.

**Comment by nomisRev** [comment]
@diesalbla thanks for the reference and reminding me off this ticket! 
I am not actively working on this so I removed myself as the assigned person.

@raulraja these tickets are still all valid, right? But higher priority would be to move current tutorial like docs to API docs?

**Comment by raulraja** [comment]
yes, this need to be implemented as kdocs under the new style of documentation inside the actual types and package files .

**Comment by nomisRev** [comment]
@raulraja also the pattern docs? This tickets is about adding a doc similar to [error handling](https://arrow-kt.io/docs/patterns/error_handling/)

**Comment by raulraja** [comment]
The patterns docs or generic docs related to optics can be written as markdown files at the package level following dokka's conventions. Those package pages are published at the root of the api docs. So yes any tutorial style doc can also be written in the kdocs or markdown as expected by dokka so it's processed in the same way as the rest of the classes.
